wandb/apis/public/api.py

```diff
@@ -424,6 +440,16 @@
             },
         )
 
+    def _send_api_request(
+        self,
+        request: ApiRequest,
+    ) -> ApiResponse:
+        if self._service is None:
```

Comment:
please leave a TODO to move it up the stack once this is requirement (i.e., all of publicapi requests are routed through wandb-core)